src/editorconfig.rs

```diff
@@ -0,0 +1,374 @@
+                    .map_or(config, |tab_width| match tab_width {
+                        TabWidth::Value(indent_width) => config.with_indent_width(indent_width),
+                    })
+            }
+        }
+    }
+    if let Ok(indent_style) = properties.get::<IndentStyle>() {
+        config = match indent_style {
+            IndentStyle::Tabs => config.with_indent_type(IndentType::Tabs),
+            IndentStyle::Spaces => config.with_indent_type(IndentType::Spaces),
+        }
+    }
+    if let Ok(max_line_length) = properties.get::<MaxLineLen>() {
+        config = match max_line_length {
+            MaxLineLen::Value(column_width) => config.with_column_width(column_width),
+            MaxLineLen::Off => config.with_column_width(0),
+        }
+    }
+    if let Ok(quote_type) = properties.get::<QuoteTypeChoice>() {
+        config = match quote_type {
+            QuoteTypeChoice::Double => config.with_quote_style(QuoteStyle::AutoPreferDouble),
+            QuoteTypeChoice::Single => config.with_quote_style(QuoteStyle::AutoPreferSingle),
+            QuoteTypeChoice::Auto => config,
+        }
+    }
+    if let Ok(call_parentheses) = properties.get::<CallParenthesesChoice>() {
+        config = match call_parentheses {
+            CallParenthesesChoice::Always => config.with_call_parentheses(CallParenType::Always),
+            CallParenthesesChoice::NoSingleString => {
+                config.with_call_parentheses(CallParenType::NoSingleString)
+            }
+            CallParenthesesChoice::NoSingleTable => {
+                config.with_call_parentheses(CallParenType::NoSingleTable)
+            }
+            CallParenthesesChoice::None => config.with_call_parentheses(CallParenType::None),
+        }
+    }
+    if let Ok(collapse_simple_statement) = properties.get::<CollapseSimpleStatementChoice>() {
+        config = match collapse_simple_statement {
+            CollapseSimpleStatementChoice::Never => {
+                config.with_collapse_simple_statement(CollapseSimpleStatement::Never)
+            }
+            CollapseSimpleStatementChoice::FunctionOnly => {
+                config.with_collapse_simple_statement(CollapseSimpleStatement::FunctionOnly)
+            }
+            CollapseSimpleStatementChoice::ConditionalOnly => {
+                config.with_collapse_simple_statement(CollapseSimpleStatement::ConditionalOnly)
+            }
+            CollapseSimpleStatementChoice::Always => {
+                config.with_collapse_simple_statement(CollapseSimpleStatement::Always)
+            }
+        }
+    }
+
+    config
+}
+
+// Extracted from ec4rs::property
+macro_rules! property_choice {
+    ($prop_id:ident, $name:literal; $(($variant:ident, $string:literal)),+) => {
+        #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+        #[repr(u8)]
+        #[doc = concat!("The [`",$name,"`](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#",$name,") property.")]
```

Comment:
```suggestion
```

Reply:
[Also removed](https://github.com/JohnnyMorganz/StyLua/pull/645/commits/550f40b7e31472e44ef787dcb2cc54fafe85ed02#diff-bde938c835cc94f24ee5b19bd4ac7eecf0c80eae46ffe2df066bdd720d4b11b1L104) `#[allow(missing_docs)]` or does it have any use?

Reply:
Don't think so, we don't deny missing docs at the crate level. Fine to remove